DEVELOPER_NOTES.md

```diff
@@ -0,0 +1,89 @@
+# Developer instructions
+
+Ready to contribute to `undate`? Here are some instructions to get you started.
+
+## Setup
+
+### Use git to checkout a copy of the repository
+```sh
+git clone git@github.com:dh-tech/undate-python.git
+cd undate-python
+```
+
+### Install and initialize git flow
+
+This repository uses [git-flow](https://github.com/nvie/gitflow) branching conventions:
+- **main** will always contain the most recent release
+- **develop** branch is the latest version of work in progress
+
+Pull requests for new features should be made against the **develop** branch.
+
+We recommended installing git-flow.
+1. On OSX, use brew or ports, e.g.: `brew install git-flow`; on Ubuntu/Debian, `apt-get install git-flow`
+2. Initialize it in your local copy of this repository: run `git flow init` and accept all the defaults.
+3. Use `git flow feature start feature-name` to create a new feature development branch.
+4. Feature branches can be merged locally with git flow or by GitHub pull request.
+4. Use git flow for releases with `git flow release start x.x.x` and `git flow release finish x.x.x`, where x.x.x is the version number for the new release.
+
+If you cannot or prefer not to install git flow, you can do the equivalent manually.
+1. Check out the develop branch: `git checkout develop`
+3. Create new feature manually from the develop branch: `git checkout -b feature/xxx-name`
+
+### Create a Python virtual environment
+
+Use a recent version of python 3. We highly recommend using a python virtualenv, e.g.
+```
+python3 -m venv undate
+source undate/bin/activate
```

Comment:
By default it looks like `undate` isn't gitignored. Do we want to change this to `.venv`, or add `undate` to `.gitignore`? `uv` uses `.venv` by default and that's in most `.gitignore` files (including ours).

It might also be helpful to either specify a Python version here (e.g. `python3.12 -m venv undate`) or include a `.pythonversion` file so folks using `pyenv` would install a virtual environment with our preferred version.

Reply:
Thanks for flagging this. I typically put my virtualenvs elsewhere and I know other tools have specific preferences.

I like the suggestions to switch to `.venv` and create a pythonversion file as a default for development.